mitodl/ocw-studio
# Finalized logic for determining target file path for WebsiteContent objects

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #261 

#### What's this PR do?
Updates the logic for determining the destination filepath for any given `WebsiteContent` object when syncing to a storage backend. Check the linked issue for a link to the doc which describes the desired behavior

#### How should this be manually tested?
- Pick some website with the OCW course site starter
- Set all `content_filepath` values to null for all `WebsiteContent` records associated with that website
- Make sure you have created a `sitemetadata` content object
- Create two content objects for Page or Resource with the same title
- Create a third content object for Page or Resource
- In Django admin, change the `dirpath` and the `filename` for that last content object you just created (and save, of course)
- Sync this website to Github

The following should be true:
- The file created for the `sitemetadata` object should exist at the path indicated by the "file" value in the site config
- The filename for the first two content object you created should be equal to the slug-ified title, with the second one having a suffix of `-2`, and both objects should be in the `content` directory
- The last content object you created should exist in Github at this path: `<your dirpath value>/<your filename value>.md`

EDIT:
Also check to make sure that these things are happening:
- All of your `WebsiteContent` records that were created for "folder"-type config items have the correct `dirpath` filled in
- Your `ContentSyncState` checksums have been updated
- The `sync_content_state` management command works as expected

#### Any background context you want to provide?
- In order to make the reviews more manageable, I created a separate issue for removing `WebsiteContent.content_filepath` (https://github.com/mitodl/ocw-studio/issues/268). This PR makes that field obsolete, but it will still exist until we remove it in that issue.
- I add the `get_destination_filepath` method in the github API module b/c it will eventually have some Github specific logic
